adapters/dbs/mongo.js

```diff
@@ -0,0 +1,384 @@
+ * @param {string} options.deleteid The mongo id of the document that should be removed
+ * @param {string} options.id If options.deleteid is provided this value is ignored - alias for options.deleteid
+ * @param  {Function} cb      Callback function for delete
+ */
+const _DELETE = function (options, cb) {
+	try {
+		let Model = options.model || this.model;
+		let deleteid = options.deleteid || options.id;
+		if (typeof deleteid !== 'string') throw new Error('Must specify "deleteid" or "id" for delete');
+		Model.remove({ _id: deleteid }, cb); 
+	}
+	catch (e) {
+		cb(e);
+	}
+};
+
+/**
+ * Convenience method for .remove mongoose method but returns the deleted document
+ * @param  {Object}   options Configurable options for mongo delete
+ * @param {Object} [options.model=this.model] The mongoose model for query will default to the this.model value if not defined
+ * @param {string} options.deleteid The mongo id of the document that should be removed
+ * @param {string} options.id If options.deleteid is provided this value is ignored - alias for options.deleteid
+ * @param  {Function} cb      Callback function for delete
+ */
+const _DELETED = function (options, cb) {
+	try {
+		_LOAD.call(this, { model: options.model, query: options.deleteid || options.id }, (err1, loaded) => {
+			if (err1) cb(err1);
+			else {
+				_DELETE.call(this, options, (err2) => {
+					if (err2) cb(err2);
+					else cb(null, loaded);
+				});
+			}
+		});
+	}
+	catch (e) {
+		cb(e);
+	}
+};
+
+/**
+ * An mongoose specific adapter which provides CRUD methods for a given model
+ * @class Mongo_Adapter
+ */
+const MONGO_ADAPTER = class Mongo_Adapter {
+	/**
+	 * @constructor
+	 * @param  {Object} [options={}] Configurable options for the mongo adapter
+	 * @param {string} options.docid Specifies the field which should be queried by default for .load
+	 * @param {Object} options.model Mongoose model that should be used in CRUD operations by default
+	 * @param {Object|string} [options.sort="-createdat"] Specifies default sort logic for .search queries
+	 * @param {number} [options.limit=500] Specifies a default limit to the total documents returned in a .search query
+	 * @param {number} [options.offset=0] Specifies a default amount of documents to skip in a .search query
+	 * @param {Object|string} [options.population] Optional population configuration for documents returned in .load and .search queries
+	 * @param {Object} [options.fields] Optional configuration for limiting fields that are returned in .load and .search queries
+	 * @param {number} [options.pagelength=15] Specifies max number of documents that should appear in each sub-set for pagination
+	 * @param {string[]} [options.xss_whitelist=false] Configuration for XSS whitelist package. If false XSS whitelisting will be ignored
+	 */
+	constructor (options = {}) {
+		this.docid = options.docid;
+		this.model = options.model;
```

Comment:
you need to be able to pass in options.modelname, which will use mongoose.model(options.modelname);